packages/ibm-products/src/components/AddSelect/AddSelectColumn.js

```diff
@@ -139,33 +151,53 @@ export let AddSelectColumn = ({
               sortByLabel={sortByLabel}
             />
             {filterByOpts.length > 0 && (
-              <OverflowMenu
-                renderIcon={(props) => <Filter size={32} {...props} />}
-                className={`${colClass}-overflow`}
-                flipped
-                aria-label={filterByLabel}
-                iconDescription={filterByLabel}
-              >
-                {filterByOpts.map((opt) => (
-                  <div
-                    key={opt}
-                    className={`${carbonPrefix}--overflow-menu-options__option`}
+              <FeatureFlags enableV12DynamicFloatingStyles>
+                <Popover
+                  align="bottom-right"
+                  isTabTip
+                  onKeyDown={(e) => {
+                    if (e.key === 'Escape') {
+                      setFilterOpen(false);
+                      e.preventDefault(); // prevent modal close
+                      e.stopPropagation();
+                      const filterBtn = document.querySelector(
+                        `#${filterBtnId}`
+                      );
+                      filterBtn?.focus(); // workaround to return focus, refs not possible on button inside popover for some reason
+                    }
+                  }}
+                  onRequestClose={() => setFilterOpen(false)}
+                  open={filterOpen}
+                >
+                  <IconButton
+                    size="md"
+                    id={filterBtnId}
+                    hasIconOnly
+                    aria-expanded={filterOpen}
+                    badgeCount={filters.length > 0 ? 0 : undefined} // setting to 0 renders an empty dot
```

Comment:
Ooh, this is a nice addition! Is _that_ why you were looking at the badge/icon indicators? 😉